### PR TITLE
Restructure the docs

### PR DESCRIPTION
**Pull request by jaidisido**
### Feature or Bugfix
- Refactoring

### Detail
- Refactored the docs structure to leverage Sphinx templates and the ability to showcase nested methods

- By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/source/_templates/custom-class-template.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{{ fullname | escape | underline}}
```
Comment (jaidisido) [inline docs/source/_templates/custom-class-template.rst:1]: Template enables generating documentation for not only classes but also their methods and attributes

File: docs/.gitignore [hunk 2]
```diff
@@ -1,3 +1,2 @@
 release/next/api/*
-source/cli/*
```
Comment (jaidisido) [inline docs/.gitignore:2]: @malachi-constant I had to remove this in order to push the `docs/source/cli/aws_ddk.rst` file. Is this required in the .gitignore? If so, can we merge this and then put it back?

Comment (malachi-constant) [reply docs/.gitignore:2]: Yes we can and should remove this now that we aren't autogenerating those source *rst files.

**Review by malachi-constant — APPROVED** [review]

**Comment by malachi-constant** [comment]
Love what you did here @jaidisido, just built it locally and it looks great.

File: core/aws_ddk_core/pipelines/stage.py [hunk 3]
```diff
@@ -21,42 +21,43 @@
 
 class Stage(Construct):
     """
-    Abstract class that represents a stage within a data pipeline.
```
Comment (kukushking) [inline core/aws_ddk_core/pipelines/stage.py:24]: Why remove this?

File: core/aws_ddk_core/resources/_kms.py [hunk 4]
```diff
@@ -33,6 +33,10 @@ class KeyConfigSchema(BaseSchema):
 
 
 class KMSFactory:
+    """
+    Class factory to create and configure Key Management Service DDK resources, including Keys.
```
Comment (kukushking) [inline core/aws_ddk_core/resources/_kms.py:37]: Class that contains factory methods to create KMS resources? Let's use the common terminology

**Pull request merged**